src/data/nat/factorization.lean

```diff
@@ -385,6 +385,82 @@ begin
   exact support_add_eq (factorization_disjoint_of_coprime hab),
 end
 
+/-! ### Factorization and `is_square` -/
+
+lemma is_square_iff_factorization_even {m : ℕ} :
+  is_square m ↔ ∀ p, even (m.factorization p) :=
+begin
+  rcases eq_or_ne m 0 with rfl | hm_zero,
+  { simp, },
+  split,
+  { rintros ⟨c, hc⟩,
+    by_cases hc' : c = 0,
+    { rw [hc', mul_zero] at hc,
+      rw hc,
+      simp only [factorization_zero, finsupp.coe_zero, pi.zero_apply, forall_const, even_zero], },
+    { intros p,
+      rw [hc, nat.factorization_mul hc' hc', finsupp.coe_add],
+      exact even_add_self _, }, },
+  { intros hp,
+    use m.factorization.prod (λ p i, p ^ (i / 2)),
+    rw ← finsupp.prod_mul,
+    conv { to_lhs, rw ← nat.factorization_prod_pow_eq_self hm_zero, },
+    apply finsupp.prod_congr,
+    intros x hx,
+    rcases hp x with ⟨c, hc⟩,
+    rw [hc, ←two_mul],
+    simp only [mul_div_right, succ_pos'],
+    rw [←pow_add, ←two_mul], },
+end
+
+lemma _root_.of_coprime_of_mul_left {m n : ℕ} (hmn : m.coprime n) :
```

Comment:
This doesn't look like a good name. It doesn't mention `is_square` at all.

Reply:
Hmm, good point. I changed the name to this at @YaelDillies's request. It was previously `is_square.of_coprime_of_mul_left` but this is in the `nat` namespace so that would mess up the dot notation.

Is the correct thing something like `_root_.is_square.of_coprime_of_mul_left` ?

Reply:
Yes indeed. Sorry if I was unclear!

Reply:
Oh not at all! I'm still learning all the nooks and crannies of the system. :-) Didn't think of the double dot at the time.

Reply:
OK, name updated. @jcommelin does that work?

Reply:
Just bumping for @jcommelin